fabric-1.19/src/main/java/org/dynmap/fabric_1_19/FabricWorld.java

```diff
@@ -0,0 +1,237 @@
+        this(plugin, getWorldName(plugin, w), w.getHeight(),
+                w.getSeaLevel(),
+                w.getRegistryKey() == World.NETHER,
+                w.getRegistryKey() == World.END,
+                w.getRegistryKey().getValue().getPath(),
+                w.getDimension().minY());
+        setWorldLoaded(w);
+    }
+
+    public FabricWorld(DynmapPlugin plugin, String name, int height, int sealevel, boolean nether, boolean the_end, String deftitle, int miny) {
+        super(name, Math.min(height, maxWorldHeight), sealevel, miny);
+        this.plugin = plugin;
+        world = null;
+        setTitle(deftitle);
+        isnether = nether;
+        istheend = the_end;
+        skylight = !(isnether || istheend);
+
+        if (isnether) {
+            env = "nether";
+        } else if (istheend) {
+            env = "the_end";
+        } else {
+            env = "normal";
+        }
+
+    }
+
+    /* Test if world is nether */
+    @Override
+    public boolean isNether() {
+        return isnether;
+    }
+
+    public boolean isTheEnd() {
+        return istheend;
+    }
+
+    /* Get world spawn location */
+    @Override
+    public DynmapLocation getSpawnLocation() {
+        if (world != null) {
+            spawnloc.x = world.getLevelProperties().getSpawnX();
+            spawnloc.y = world.getLevelProperties().getSpawnY();
+            spawnloc.z = world.getLevelProperties().getSpawnZ();
+            spawnloc.world = this.getName();
+        }
+        return spawnloc;
+    }
+
+    /* Get world time */
+    @Override
+    public long getTime() {
+        if (world != null)
+            return world.getTimeOfDay();
+        else
+            return -1;
+    }
+
+    /* World is storming */
+    @Override
+    public boolean hasStorm() {
+        if (world != null)
+            return world.isRaining();
+        else
+            return false;
+    }
+
+    /* World is thundering */
+    @Override
+    public boolean isThundering() {
+        if (world != null)
+            return world.isThundering();
+        else
+            return false;
+    }
+
+    /* World is loaded */
+    @Override
+    public boolean isLoaded() {
+        return (world != null);
+    }
+
+    /* Set world to unloaded */
+    @Override
+    public void setWorldUnloaded() {
+        getSpawnLocation();
+        world = null;
+    }
+
+    /* Set world to loaded */
+    public void setWorldLoaded(World w) {
+        world = w;
+        this.sealevel = w.getSeaLevel();   // Read actual current sealevel from world
+        // Update lighting table
+        for (int i = 0; i < 16; i++) {
+            this.setBrightnessTableEntry(i, FabricWorld.getBrightness(w.getDimension(), i));
+        }
+    }
+
+    public static float getBrightness(DimensionType type, int lightLevel) {
+        float f = (float) lightLevel / 15.0F;
+        float g = f / (4.0F - 3.0F * f);
+        return MathHelper.lerp(type.ambientLight(), g, 1.0F);
+    }
```

Comment:
getBrightness function doesn't exist anymore so we took the implementation from Minecraft 1.18.2